src/index.js

```diff
@@ -0,0 +1,130 @@
+// Constants
+const BASE_URL = 'http://127.0.0.1:5000';
+const DEFAULT_CITY = 'Los Angeles';
+const increaseTemp = document.getElementById('increaseTempControl');
+const decreaseTemp = document.getElementById('decreaseTempControl');
+const tempValue = document.getElementById('tempValue');
+const landscape = document.getElementById('landscape');
+const cityNameInput = document.getElementById('cityNameInput');
+const cityNameReset = document.getElementById('cityNameReset'); 
+const headerCityName = document.getElementById('headerCityName');
+const currentTempButton = document.getElementById('currentTempButton');
+
+// Changes color + landscape by Temperature
+let temp = 60;
+const changeTemp = () => {
+    let landscapeEmojis = '';
+    if (temp >= 80) {
+        tempValue.style.color = 'red';
+        landscapeEmojis = '🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂';
+    } else if (temp< 80 && temp >= 70) {
+        tempValue.style.color = 'orange';
+        landscapeEmojis = '🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷';
+    } else if (temp < 70 && temp >= 60) {
+        tempValue.style.color = 'yellow';
+        landscapeEmojis = '🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃';
+    } else if (temp < 60 && temp >= 50) {
+        tempValue.style.color = 'green';
+        landscapeEmojis = '🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲';
+    } else {
+        tempValue.style.color = 'teal';
+        landscapeEmojis = '🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲';
+    }
+    landscape.textContent = landscapeEmojis;
+};
+
+// Changes sky by selection
+const changeSky = () => {
+    const selectedSky = document.getElementById('skySelect').value;
+    const theSky = document.getElementById('sky');
+    let sky = '🌞😎🌞';
+    if (selectedSky === 'sunny') {
+        sky = '🌞😎🌞';
+    } else if (selectedSky === 'cloudy') {
+        sky = '🌥️🌥️🌥️';
+    } else if (selectedSky === 'rainy') {
+        sky = '🌧️🌧️🌧️';
+    } else if (selectedSky === 'snowy') {
+        sky = '🌨️🌨️🌨️';
+    }
+    theSky.textContent = sky;
+};
+
+// Event Handler Functions
+const kelvintoFahrenheit = (temp) => Math.floor((temp - 273.15) * 1.8 + 32);
+
+const incTempButton = () => {
+   temp++;
+   tempValue.textContent = `${temp}`; 
+};
+
+const decTempButton = () => {
+    temp--;
+    tempValue.textContent = `${temp}`;
+};
+
+const updateHeaderCityName = () => {
+    headerCityName.textContent = cityNameInput.value;
+};
+
+const resetCityName = () => {
+    startUI();
+};
+
+const getTemp = () => {
+    // Get city name from input field
+    const cityName = cityNameInput.value;
+
+    // Make API call to fetch location information 
+    axios.get(`${BASE_URL}/location?q=${cityName}`)
```

Comment:
👀 Prefer to send query parameters using an option object rather than coding them directly into the URL. If we pass them as part of an object, axios knows they should be query parameters and can take care of encoding them if necessary.

```js
    axios.get(`${BASE_URL}/location`, {params: {q: cityName}})
```

Reply:
Remember that in order to make use of promises in helper functions, it's best to `return` the promise chain, which allows the caller to chain on additional operations after the async operation has completed if needed (`get` returns a `Promise` to an axios `response`). This would be essential in order to split the temperature and weather calls into two distinct helpers, use those two helpers to build a function that gets the weather for a city, then finally perform the updates to the page after the full async lookup has completed.